datafusion/physical-plan/src/aggregates/row_hash.rs

```diff
@@ -1060,18 +1084,26 @@ impl GroupedHashAggregateStream {
         Ok(Some(batch))
     }
 
+    /// Determines if `spill_state_if_oom` can free up memory by spilling state to disk
+    fn can_spill_on_oom(&self) -> bool {
```

Comment:
nit would be to move this closer in the code to `can_emit_early_on_oom` so they are closer together in the code